### ISSUE TITLE
Allow Columns to be Resized on Table View for Keyboard-Only Users

### ISSUE THREAD

**Issue by MRichards99** [ral-facilities/datagateway#769]
**Description**:
[WCAG 2.1.1](https://www.w3.org/TR/WCAG21/#keyboard) states that all functionality needs to be operable when only using a keyboard. One piece of functionality which requires a mouse is resizing columns on table view. This criteria is level A, which is basically level 1 of 3 in terms of the standard's strictness - level A is the least strict of the levels so I guess this means this is high priority in terms of the standards? I.e. you can't go and meet level AAA if we can't meet level A.

I found one example of how this is done using arrow keys - https://salesforce-ux.github.io/dnd-a11y-patterns/#/resize?_k=1l94ho. That URL came from a [long-ish article on the subject](https://medium.com/salesforce-ux/4-major-patterns-for-accessible-drag-and-drop-1d43f64ebf09), basically suggesting this can be implemented using `<input type="range />` sliders.

@louise-davies suggested this could present problems for screen readers who expect a slider to be an actual slider and not for resizing - I've never used a screen reader but I think I'll need to use one to see how DataGateway is like when using one. When I do this, I'll try and consider this point while doing that.

I can see how this might be tricky to implement, so this might require further discussion on a) how this could be implemented and b) is this worth implementing/do we have the developer effort to do so.

This issue has been opened as a result of #738 

**Acceptance criteria**:
- [ ] Columns throughout table view should be able to be resized using a keyboard, with no need for a mouse
- [ ] If any non-standard shortcuts are needed for this functionality (tab, space, arrow keys are all 'standard'), these should be clearly visible for the user

**Comment by louise-davies** [comment]
maybe it's worth investigating other datatables applications (i.e. excel online or google sheets etc)

**Comment by MRichards99** [comment]
A quick investigation between Goel, Sam and I found that we couldn't resize columns on either Excel or Google Sheets using only a keyboard. Google Sheets does have an option to resize a column to a specific column width (reachable by the dropdown menu on each column) but I had to use my mouse to access this.
